op-up/main_test.go

```diff
@@ -0,0 +1,44 @@
+package main
+
+import (
+	"context"
+	"io"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/ethereum-optimism/optimism/op-devstack/sysgo"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/stretchr/testify/require"
+)
+
+func TestRun(t *testing.T) {
+	var wg sync.WaitGroup
+	defer wg.Wait()
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		require.NoError(t, run(ctx, []string{"op-up", "--dir", t.TempDir()}, io.Discard, io.Discard))
```

Comment:
You should generally avoid using `require` in a goroutine like this, since `require` will call `t.FailNow()` under the hood which won't properly stop the test thread if called anywhere other than the main test goroutine.

Instead, make an error channel for this goroutine and `select` on it i like oyu do below. This moves error handling into the main test thread.